packages/user/src/tokens.ts

```diff
@@ -0,0 +1,352 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { AccordionPanel, Menu } from '@lumino/widgets';
+import { ISignal } from '@lumino/signaling';
+import { Token } from '@lumino/coreutils';
+
+/**
+ * An ID to track the user on StateDB.
+ */
+export const USER = '@jupyterlab/user:userDB';
+
+/**
+ * The user token.
+ *
+ * NOTE: Requirer this token in your extension to access the
+ * current connected user information.
+ */
+export const ICurrentUser = new Token<IUser>('@jupyterlab/user:ICurrentUser');
+
+/**
+ * The user menu token.
+ *
+ * NOTE: Requirer this token in your extension to access the user menu
+ * (top-right menu in JupyterLab's interface).
+ */
+export const IUserMenu = new Token<Menu>('@jupyterlab/user:IUserMenu');
+
+/**
+ * The user panel token.
+ *
+ * NOTE: The left side bar panel. Useful to add user related widgets.
+ */
+export const IUserPanel = new Token<AccordionPanel>(
+  '@jupyterlab/user:IUserPanel'
+);
+
+/**
+ * An interface describing an user.
+ */
+export interface IUser {
```

Comment:
Do we really want to list attributes like `gender`, `birthDate`, `honorificPrefix`,... ?
I would suggest extending on ReadOnlyJSONObject and define only what is required (name, username,...) or used by the code (cursor position).

Moreover it seems duplicated of type `IUser.User`. So you should probably linked the two: `interface IUser extends IUser.User`.

Reply:
> Do we really want to list attributes like gender, birthDate, honorificPrefix,... ?

I tried to make it as generic as possible selecting some of the attributes listed in https://schema.org/Person that I thought could be useful for others.
But as you said, maybe its better to extend from ReadOnlyJSONObject and everyone can define whatever they need.

Reply:
I agree with @fcollonval that we should keep it very simple (`id`, `name`, `color`) and then add what the code needs (`cursor`, and maybe more, like `currentNotebook`, `currentCell`...)